File: 1518-water-bottles/1518-water-bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        total=0
        remain=0
        while(numBottles>0):
            total+=numBottles
            remain+=numBottles
            numBottles=remain//numExchange
            remain%=numExchange
        return total

File: 1518-water-bottles/1518-water-bottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int bot=numBottles, exc=numExchange;
        int total=bot, cur=bot;
        while(cur>=exc){
            total+=(cur/exc);
            cur=(cur/exc)+(cur%exc);
        }
        return total;
    }
}